.github/workflows/deploy-docker.yml

```diff
@@ -0,0 +1,58 @@
+# Builds the Docker image and publishes to the GitHub container registry
+
+name: Publish Docker Image
+
+on:
+  push:
+    branches: ["main"]
+    # Publish semver tags as releases.
+    tags: ["v*.*.*"]
+
+env:
+  # TODO: Also publish to Codeberg
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Log in to container registry
+        # docker/login-action@v3.3.0
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract Docker metadata
+        id: meta
+        # docker/metadata-action@v5.5.1
+        uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81
+        with:
+          # e.g. 'ghcr.io/byu-cs-discord/csbot'
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+
+      - name: Build and push Docker image
+        # docker/build-push-action@v6.7.0
+        uses: docker/build-push-action@5cd11c3a4ced054e52742c5fd54dca954e0edd85
+        with:
+          context: .
+          file: ./Dockerfile
```

Comment:
Important to note that our current Dockerfile was designed to be a standardized dev environment (for everyone using Dev Containers in vscode), not a production deployment. It should probably be rewritten with the new purpose in mind.

Reply:
Kk, I'll get started straightening that out

Reply:
Ok that should to the trick: Separate Dockerfile for dev and prod, and I actually tested the prod one this time lol.

It auto-creates the database for us when using Docker! How nice